wiring/inc/spark_wiring_watchdog.h

```diff
@@ -108,5 +108,64 @@ inline void application_checkin() { ApplicationWatchdog::checkin(); }
 
 inline void application_checkin() {  }
 
-#endif
+#endif // PLATFORM_THREADING
+
+
+#if Wiring_Watchdog
+
+#include "watchdog_hal.h"
+
+namespace particle {
+
+enum class WatchdogCaps : uint32_t {
+    NONE            = WATCHDOG_CAPS_NONE,
+    RESET           = WATCHDOG_CAPS_RESET,
+    INT             = WATCHDOG_CAPS_INT,
+    RECONFIGURABLE  = WATCHDOG_CAPS_RECONFIGURABLE,
+    STOPPABLE       = WATCHDOG_CAPS_STOPPABLE,
+    ALL             = WATCHDOG_CAPS_ALL
+};
+
+typedef std::function<void(void)> WatchdogOnExpiredStdFunction;
+typedef void (*WatchdogOnExpiredCallback)(void);
```

Comment:
Context.

Reply:
We could also keep this and have users bind all parameters, which is better C++ than passing and casting `void *`